airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java

```diff
@@ -76,6 +78,14 @@ public DatabaseMetaData getMetaData() throws SQLException {
     try (final Connection connection = dataSource.getConnection()) {
       final DatabaseMetaData metaData = connection.getMetaData();
       return metaData;
+    } catch (final SQLException e) {
+      // Some databases like Redshift will have null cause
+      if (Objects.isNull(e.getCause())) {
+        throw new ConnectionErrorException(e.getSQLState(), e.getErrorCode(), e.getMessage(), e);
+      } else {
+        final SQLException cause = (SQLException) e.getCause();
```

Comment:
This can throw a `ClassCastException`, because e.getClause() returns throwable and it does not have to be a `SQLException`. Furthermore, I don't see a need to cast cause to `SQLException`, because `ConnectionErrorException` takes `Throwable` as the last argument.

Reply:
To my understanding the reason for the cast to `SQLException` is because `SQLException` has the method `getErrorCode` but `Throwable` does not since it's a super of `SQLException`. This seems reasonable _albeit_ not pretty since the error code is valuable to diagnosing

Reply:
Right, `e` is `SQLException` and then `e` has a cause that's another `Throwable`. I am not sure why we are getting the error code and message from `cause` and not from `e`.

Reply:
One way to address this would be to change line 83 to
`if (Objects.isNull(e.getCause()) || !(e.getCause() instanceof SQLException)) {`

Reply:
I remember now from my conversation with Andrii that the reason for getting information from `cause` instead of `e` is because this logic is to get the root cause information. This does, however introduce the `ClassCastException` as you mentioned so this may be better with simply getting the information from the top level exception to avoid other bugs when the underlying cause wasn't `SQLException`